ark/internal/genai/memory.go

```diff
@@ -50,9 +51,11 @@ type Config struct {
 }
 
 type MessagesRequest struct {
-	SessionID string                                   `json:"session_id"`
-	QueryID   string                                   `json:"query_id"`
-	Messages  []openai.ChatCompletionMessageParamUnion `json:"messages"`
+	SessionID   string                                   `json:"session_id"`
+	QueryID     string                                   `json:"query_id"`
+	Messages    []openai.ChatCompletionMessageParamUnion `json:"messages"`
+	AgentPrompt string                                   `json:"agent_prompt,omitempty"`
```

Comment:
I don't think we need the agent prompt here or stored separately cause it is in the `assistant` message so should be claer